src/couch/src/couch_doc.erl

```diff
@@ -130,7 +131,8 @@ from_json_obj_validate(EJson) ->
     from_json_obj_validate(EJson, undefined).
 
 from_json_obj_validate(EJson, DbName) ->
-    MaxSize = config:get_integer("couchdb", "max_document_size", 4294967296),
+    ConfigMaxSize = config:get_integer("couchdb", "max_document_size", 4294967296),
```

Comment:
Should we use `config:get_integer("couchdb", "max_document_size", ?DOCUMENT_SIZE_LIMIT)` to get rid of magic number in the codebase?

Reply:
I was trying to iron out a bug last night where changing the magic number to `?DOCUMENT_SIZE_LIMIT` broke the tests. I am not sure if it is flaky tests or a bug. I was getting a lot of inconsistent results last night.

Reply:
> I was trying to iron out a bug last night where changing the magic number to ?DOCUMENT_SIZE_LIMIT broke the tests. 

Did you try modifying this mock https://github.com/apache/couchdb/blob/main/src/couch/test/eunit/couch_doc_json_tests.erl#L42 ?

Reply:
I think it should be `fun("couchdb", "max_document_size", _) -> 1024 end),`

Reply:
No, I did not. I assume something similar should be done here. https://github.com/apache/couchdb/blob/32179742424388ef3cd0a088aa93d9c12099814f/src/couch/test/eunit/couch_doc_tests.erl#L101.

Reply:
Okay, looks like https://github.com/apache/couchdb/pull/3632/commits/fbd84fe72734a44bc61b9e7c2f9cc95548252fed fixed it.